feat: Update the minimum password length.

feanil opened the pull request:
This changes:

* The default minimum length of passwords from 2 characters to 8 characters. 
* [Updates the CMS password settings](https://github.com/openedx/edx-platform/pull/33373/files#diff-a16fd1c6cf75bdebdf97a2633b533d39e43975c6e98af58787ed6af1282eca71) so that they pull the defaults from the LMS rather than have their own copy.

Operational Impact: None

If you have an existing password, this change along will not force you to update it.  However if you reset your password or go to change it, you'll have to conform to the new guidelines.

If you would like to force people to update their password, you'll probably want to take a look at [the password_policy plugin and its settings](https://github.com/openedx/edx-platform/blob/2033dcf6ace133719aaeb72dc5dd6ee521a7ac42/openedx/core/djangoapps/password_policy/settings/common.py#L13).

feanil committed:
feat: Update the minimum password length.

feanil committed:
test: Fix validation tests.

Make them resilient to the default changing where it makes sense.

feanil committed:
test: Update more tests that had short passwords.

feanil commented on cms/envs/common.py:
```diff
@@ -121,6 +121,9 @@
     _make_locale_paths,
+
+    # Password Validator Settings
+    AUTH_PASSWORD_VALIDATORS
```
This is a critical functional change.  This will cause the CMS and LMS to have the same password policies by default.

magajh replied:
Thanks @feanil! Should we include this in the PR description? It seems like a key piece of information that could be helpful to have at a glance.

feanil replied:
Updated the description.

feanil committed:
test: Update to an even longer password.

alangsto commented on cms/envs/common.py:
```diff
@@ -121,6 +121,9 @@
     _make_locale_paths,
+
+    # Password Validator Settings
+    AUTH_PASSWORD_VALIDATORS
```
Just curious about leaving this setting in if it has no assigned value - should we just remove it?

feanil replied:
This pulls this setting from the LMS so the LMS and CMS have the same setting instead of having a copy over here.

pshiu commented on cms/envs/common.py:
```diff
@@ -121,6 +121,9 @@
     _make_locale_paths,
+
+    # Password Validator Settings
+    AUTH_PASSWORD_VALIDATORS
```
[log] This introduces a change making the CMS `MinimumLengthValidator` default to 8 instead of 2, which is desired behavior if I understand correctly!

pshiu commented on lms/djangoapps/courseware/testutils.py:
```diff
@@ -81,7 +81,7 @@ def login(self):
         Logs in the test user.
         """
-        self.client.login(username=self.user.username, password='test')
+        self.client.login(username=self.user.username, password='Password1234')
```
[global] Why not `TEST_PASSWORD`?

feanil replied:
`TEST_PASSWORD` was not available from the superclass for this test class so I just did a manual update.

pshiu commented on lms/envs/common.py:
```diff
@@ -3756,7 +3756,7 @@ def _make_locale_paths(settings):  # pylint: disable=missing-function-docstring
         "NAME": "common.djangoapps.util.password_policy_validators.MinimumLengthValidator",
         "OPTIONS": {
-            "min_length": 2
+            "min_length": 8
```
[log] Change of default of `MinimumLengthValidator` from 2 to 8. Also checked against our config at edx.org, and concur in this not causing a breaking change.

pshiu reviewed (approved):
Thanks, @feanil!

magajh commented on lms/djangoapps/mobile_api/testutils.py:
```diff
@@ -50,7 +50,7 @@ def setUp(self):
         )
         self.user = UserFactory.create()
-        self.password = 'test'
+        self.password = 'Password1234'
```
Since this class inherits from `ModuleStoreTestCase`, could we use `TEST_PASSWORD`?

magajh commented on common/djangoapps/student/tests/factories.py:
```diff
@@ -81,7 +81,7 @@ class Meta:
         django_get_or_create = ('email', 'username')
 
-    _DEFAULT_PASSWORD = 'test'
+    _DEFAULT_PASSWORD = 'Password1234'
```
Is it possible to reuse the `TEST_PASSWORD` constant here? It might help in maintaining consistency

feanil replied:
There are a few places, where I think making new accesses to the TEST_PASSWORD variable would result in weird coupling so I left them as is.  I agree that we could make further improvements here but I think that what I have done so far is an improvement and good enough for now.

magajh commented on lms/djangoapps/survey/tests/test_utils.py:
```diff
@@ -28,7 +28,7 @@ def setUp(self):
 
         # Create two accounts
-        self.password = 'abc'
+        self.password = 'Password1234'
```
I think we can use the `TEST_PASSWORD` attribute here

magajh commented on lms/djangoapps/survey/tests/test_views.py:
```diff
@@ -29,7 +29,7 @@ def setUp(self):
 
         # Create two accounts
-        self.password = 'abc'
+        self.password = 'Password1234'
```
Since this class inherits from `ModuleStoreTestCase`, can we use the `TEST_PASSWORD` attribute?

magajh commented on openedx/tests/xblock_integration/test_crowdsource_hinter.py:
```diff
@@ -21,8 +21,8 @@ class TestCrowdsourceHinter(SharedModuleStoreTestCase, LoginEnrollmentTestCase):
     STUDENTS = [
-        {'email': 'view@test.com', 'password': 'foo'},
-        {'email': 'view2@test.com', 'password': 'foo'}
+        {'email': 'view@test.com', 'password': 'Password1234'},
```
Maybe we could reuse the `TEST_PASSWORD` attribute here for consistency?

magajh commented on lms/djangoapps/discussion/rest_api/tests/test_views.py:
```diff
@@ -929,7 +929,7 @@ class CourseTopicsViewV3Test(DiscussionAPIViewTestMixin, CommentsServiceMockMixi
     def setUp(self) -> None:
         super().setUp()
-        self.password = "password"
+        self.password = self.TEST_PASSWORD
```
Would it be possible to just use `TEST_PASSWORD` and remove the `password` attribute from these setup classes? I noticed the same on lines [2761](https://github.com/openedx/edx-platform/pull/33373/files#diff-9621c3aa4d1648b5f872e89899a234ee595002e2b69b72211361c9c61741b825R2761) and [3059](https://github.com/openedx/edx-platform/pull/33373/files#diff-9621c3aa4d1648b5f872e89899a234ee595002e2b69b72211361c9c61741b825R3059).

feanil replied:
Definitely possible but a lot more work so leaving this as a future cleanup.

magajh replied:
Agree. Where should we leave a note or reminder to perform this cleanup in the future? Should we create an issue for this task?

magajh commented on cms/djangoapps/contentstore/tests/test_i18n.py:
```diff
@@ -184,7 +184,7 @@ def setUp(self):
         self.uname = 'testuser'
         self.email = 'test+courses@edx.org'
-        self.password = 'foo'
+        self.password = 'Password1234'
```
Since this test case inherits from `ModuleStoreTestCase`, could we use `TEST_PASSWORD`?

magajh commented on lms/djangoapps/course_blocks/transformers/tests/helpers.py:
```diff
@@ -50,7 +50,7 @@ def setUp(self):
         super().setUp()
         # Set up users.
-        self.password = 'test'
+        self.password = 'Password1234'
```
Since this class inherits from `ModuleStoreTestCase`, could we use the `TEST_PASSWORD` attribute?

magajh commented on lms/djangoapps/course_blocks/transformers/tests/helpers.py:
```diff
@@ -253,7 +253,7 @@ def setUp(self):
                     update_block(parent_block)
 
-        self.password = 'test'
+        self.password = 'Password1234'
```
same here, I believe we can use `TEST_PASSWORD`

magajh commented on lms/djangoapps/course_wiki/tests/tests.py:
```diff
@@ -24,7 +24,7 @@ def setUp(self):
         self.student = 'view@test.com'
         self.instructor = 'view2@test.com'
-        self.password = 'foo'
+        self.password = 'Password1234'
```
Since this class inherits from `ModuleStoreTestCase`, could we use the `TEST_PASSWORD` attribute?

magajh commented on lms/djangoapps/courseware/tests/test_submitting_problems.py:
```diff
@@ -154,7 +154,7 @@ def setUp(self):
         self.course = CourseFactory.create(display_name=self.COURSE_NAME, number=self.COURSE_SLUG)
         self.student = 'view@test.com'
-        self.password = 'foo'
+        self.password = 'Password1234'
```
Since this class inherits from `ModuleStoreTestCase`, can we use `TEST_PASSWORD`?

magajh commented on lms/djangoapps/grades/rest_api/v1/tests/test_views.py:
```diff
@@ -548,7 +548,7 @@ class CourseSubmissionHistoryWithDataTest(TestSubmittingProblems):
         super().setUp()
         self.namespaced_url = 'grades_api:v1:submission_history'
-        self.password = 'test'
+        self.password = 'Password1234'
```
This test case inherits from `TestSubmittingProblems`, so I believe we can use the `TEST_PASSWORD`attribute

magajh commented on lms/djangoapps/verify_student/tests/test_views.py:
```diff
@@ -122,8 +122,8 @@ def test_start_new_verification(self):
         Test the case where the user has no pending `PhotoVerificationAttempts`,
```
I think it would be better to define 'Password1234' as a single variable and reuse it throughout this file. If not, maybe we can define the attribute in the setup classes for each necessary test case to reduce repetition.

magajh commented on openedx/tests/xblock_integration/xblock_testcase.py:
```diff
@@ -282,9 +282,9 @@ class XBlockStudentTestCaseMixin:
-        {'email': 'alice@test.edx.org', 'password': 'foo'},
-        {'email': 'bob@test.edx.org', 'password': 'foo'},
-        {'email': 'eve@test.edx.org', 'password': 'foo'},
+        {'email': 'alice@test.edx.org', 'password': 'Password1234'},
+        {'email': 'bob@test.edx.org', 'password': 'Password1234'},
+        {'email': 'eve@test.edx.org', 'password': 'Password1234'},
```
How about introducing a variable here to avoid repetition?

feanil replied:
Given how localized all the values are I don't think it's super valuable in this case.

magajh commented:
Hey @feanil, should we update the password in this mixins file for tests as well? https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/course_api/tests/mixins.py#L11. Do you think it's necessary, or is it fine to leave it as is? I'm wondering why the tests that are using the mixin aren't affected.

feanil commented:
> Hey @feanil, should we update the password in this mixins file for tests as well? https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/course_api/tests/mixins.py#L11. Do you think it's necessary, or is it fine to leave it as is? I'm wondering why the tests that are using the mixin aren't affected.

Good catch @magajh it looks like the reason this doesn't fail is because the API endpoints that are being tested don't require authentication so the fact that auth failed doesn't matter.  I've updated the password anyway so that it's less confusing.

feanil committed:
test: Update to reuse variables in more places.

Some of the places where we had explicit copies of the password were not
necessary so we referece the exsting TEST_PASSWORD variable where
possible.

magajh reviewed (approved):
LGTM!
